### PR TITLE
feat(cli): VPC context provider looks up RouteTable IDs

### LINKED ISSUE #2487: Add route table IDs to VPC context provider
I would like to create Route resources on an existing VPC. 

My use case looks something like this:
```
const new_vpc = new ec2.VpcNetwork(this, 'new_vpc' {
  // ...
});
const vpcId =  'vpc-xxx';
const existing_vpc = ec2.VpcNetwork.importFromContext(this, 'existing_vpc', {
  vpcId: vpcId,
});
const pcx = new ec2.CfnVPCPeeringConnection(this, 'new_vpc-existing_vpc', {
  vpcId: vpcId,
  peerVpcId: new_vpc.vpcId,
});
import_vpc.publicSubnets.forEach((subnet,index) => {
  new ec2.CfnRoute(this, 'route-' + index, {
    routeTableId: subnet.routeTableId,
    destinationCidrBlock: new_vpc.cidr,
    vpcPeeringConnectionId: pcx.ref,
  });
});
```

### LINKED ISSUE #2926: Subnets routeTableId is undefined when VPC was imported via fromLookup
**Describe the bug**
When importing a VPC via `ec2.Vpc.fromLookup` the containing Subnets (ISubnet) routeTableId Property is `undefined` even if there is a RouteTables associated with the specific Subnet.

**To Reproduce**
Create a VPC in your account with a Subnet and an associated RouteTable and import the VPC via `fromLookup` 

**Expected behavior**
`ISubnet.routeTableId` should contain the RouteTabeId from within the AWS account.

**Env**
OS: MacOS
CDK: 0.34.0

### PR DESCRIPTION
**Pull request by RomainMuller**
Import RouteTableIds for subnets when looking up a VPC from the `VpcContextProvider`.
This associates the Subnet-specific route table ID if there is one, and falls back to the
VPC's main route table otherwise.

Fixes #2487 
Fixes #2926 

----

Please read the [contribution guidelines](https://github.com/awslabs/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/alexa-ask/package-lock.json [hunk 1]
```diff
@@ -1802,28 +1802,28 @@
 							"dependencies": {
 								"abbrev": {
 									"version": "1.1.1",
-									"resolved": "",
+									"resolved": false,
```
[the remark this answers is not in the document]
Comment (RomainMuller) [reply packages/@aws-cdk/alexa-ask/package-lock.json]: I had to re-generate the lock files to fix the merge conflicts...

File: packages/@aws-cdk/aws-ec2/lib/util.ts [hunk 2]
```diff
@@ -40,33 +40,53 @@ export function subnetId(name: string, i: number) {
...
 
     if (Math.floor(this.groups) !== this.groups) {
       // tslint:disable-next-line:max-line-length
       throw new Error(`Amount of ${idField} (${this.subnetIds.length}) must be a multiple of availability zones (${this.availabilityZones.length}).`);
     }
+    if (this.routeTableIds.length !== this.subnetIds.length) {
+      if (routeTableIds == null) {
+        // Maintaining backwards-compatibility - this used to not be provided by the VPC Context Provider!
+        // tslint:disable-next-line: max-line-length
+        this.warning = `No routeTableIds were provided for subnets ${this.subnetIds.join(', ')}! Calling .routeTableId on these subnets will return undefined/null!`;
```
[the remark this answers is not in the document]
Comment (RomainMuller) [reply packages/@aws-cdk/aws-ec2/lib/util.ts]: That'd be a breaking change...

File: packages/@aws-cdk/aws-ec2/lib/util.ts [hunk 3]
```diff
@@ -40,33 +40,53 @@ export function subnetId(name: string, i: number) {
...
+        // tslint:disable-next-line: max-line-length
+        throw new Error(`Amount of ${routeTableIdField} (${this.routeTableIds.length}) must be equal to the amount of ${idField} (${this.subnetIds.length}).`);
+      }
+    }
 
     this.names = this.normalizeNames(names, defaultSubnetName(type), nameField);
   }
 
   public import(scope: cdk.Construct): ISubnet[] {
+    if (this.warning && this.subnetIds.length > 0) {
```
[the remark this answers is not in the document]
Comment (RomainMuller) [reply packages/@aws-cdk/aws-ec2/lib/util.ts]: Ah ! That was already the case, I was being overly cautious :)

File: packages/@aws-cdk/aws-ec2/lib/vpc.ts [hunk 4]
```diff
@@ -1342,8 +1368,20 @@ class ImportedSubnet extends Resource implements ISubnet, IPublicSubnet, IPrivat
   constructor(scope: Construct, id: string, attrs: SubnetAttributes) {
     super(scope, id);
 
+    if (!attrs.routeTableId) {
+      const ref = Token.isUnresolved(attrs.subnetId)
+        ? `at '${scope.node.path}/${id}'`
+        : `'${attrs.subnetId}'`;
+      // tslint:disable-next-line: max-line-length
+      scope.node.addWarning(`No routeTableId was provided to the subnet ${ref}. Attempting to read it's .routeTable.routeTableId will return null/undefined. (More info: https://github.com/aws/aws-cdk/pull/3171)`);
```
Comment (jordan-brough) [inline packages/@aws-cdk/aws-ec2/lib/vpc.ts:1376]: Note: For anyone directed to this PR by the `No routeTableId was provided to the subnet` warning message: See this issue: https://github.com/aws/aws-cdk/issues/19786, which about making this warning only appear when necessary.

**Comment by JoshMcCullough** [comment]
Why, when fetching a subnet via lookup, is the `routeTableId` not populated??

File: packages/@aws-cdk/aws-ec2/lib/vpc.ts (hunk 4 above)
Comment (JoshMcCullough) [reply packages/@aws-cdk/aws-ec2/lib/vpc.ts:1376]: Thanks, but how do I fix the actual warning? I need the route table ids to be pulled back so I can make use of them. Do I have to hard-code them in my config and use `from_attributes( ... routeTableId ...)`?

Comment (JoshMcCullough) [reply packages/@aws-cdk/aws-ec2/lib/vpc.ts:1376]: I ended up fetching the subnets and route tables and marrying them via `Subnet.fromSubnetAttributes`. Not ideal, but I'm sure there's a reason...

Comment (simonbuchan) [reply packages/@aws-cdk/aws-ec2/lib/vpc.ts:1376]: From linked issue, do something like:

```ts
      cdk.Annotations.of(subnet).acknowledgeWarning(
        "@aws-cdk/aws-ec2:noSubnetRouteTableId",
        "Will not read route table ID for subnet",
      );
```

**Pull request merged**